testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/DeleteAccountTest.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.testsuite.ui.account2;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+import org.jboss.arquillian.graphene.page.Page;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.keycloak.events.EventType;
+import org.keycloak.models.AccountRoles;
+import org.keycloak.representations.idm.ClientRepresentation;
+import org.keycloak.representations.idm.RequiredActionProviderRepresentation;
+import org.keycloak.representations.idm.RoleRepresentation;
+import org.keycloak.testsuite.AssertEvents;
+import org.keycloak.testsuite.admin.ApiUtil;
+import org.keycloak.testsuite.ui.account2.page.AIALoginPage;
+import org.keycloak.testsuite.ui.account2.page.AbstractLoggedInPage;
+import org.keycloak.testsuite.ui.account2.page.DeleteAccountActionConfirmPage;
+import org.keycloak.testsuite.ui.account2.page.DeleteAccountPage;
+import org.keycloak.testsuite.ui.account2.page.WelcomeScreen;
+
+/**
+ * @author Zakaria Amine <zakaria.amine88@gmail.com>
+ */
+public class DeleteAccountTest extends BaseAccountPageTest {
```

Comment:
As I [mentioned earlier](https://github.com/keycloak/keycloak/pull/7176#discussion_r467919386), I don't like that you need to do generic actions (enable/disable required action, remove role, ...) in each test. Could we move at least some of them to `@Before`/`@After` or override `addTestRealms`?

Reply:
sure thing